### ISSUE TITLE
ExtensionHost is null on EventBroker.Handlers.OnUserInterface and OnUserInterfaceAsync

### ISSUE THREAD

**Issue by BrunoJuchli** [appccelerate/appccelerate#3]
This is a copy of the issue #28 for bbv.common.EventBroker: https://github.com/bbvcommon/bbv.Common/issues/28

Upon code examination, the issue still seems to exist in Appccelerate

In both classes (OnUserInterface & OnUserInterfaceAsync), the implementation of 

``` csharp
public virtual void Initialize(object subscriber, MethodInfo handlerMethod, IExtensionHost extensionHost);
```

does not call

``` csharp
base.Initialize(subscriber, handlerMethod, extensionHost);
```

Thus, the ExtensionHost property is always null. Calls to base.HandleSubscriberMethodException(...) subsequently fail with a NullReferenceException.

For the OnBackground.cs handler, the issue does not apply, as it does not override the initialize method.

The 'pull request' would be to add the base.Initialize to the two Handlers, and add a unit test which tests the ExtensionHost property to be the one passed by Initialize :)

@ Dani:
Ich nehme an, es wird hier keinen Fix mehr für bbv.common.EventBroker geben, oder?

**Comment by ursenzler** [comment]
Will be fixed in bbv.Common, too.

**Comment by BrunoJuchli** [comment]
Glad to hear, although we'll hopefully migrate to appccelerate very soon. Thank you Urs! regards

**Comment by ursenzler** [comment]
You can find a migration how-to on appccelerate.com.

I can apply the fix not before next week.

If you need a quick fix, you can derive your own Factory from
StandardFactory and pass it to the constructor of the EventBroker and
override CreateHandler and return instances of classes with your fix.

Cheers,
Urs

Am 12.08.2012 um 10:37 schrieb BrunoJuchli notifications@github.com:

Glad to hear, although we'll hopefully migrate to appccelerate very soon.
Thank you Urs! regards

—
Reply to this email directly or view it on
GitHubhttps://github.com/appccelerate/appccelerate/issues/3#issuecomment-7673779.

**Comment by Aretnorp** [comment]
Glad to see this issue is being addressed and I am looking forward to further developments =).

**Comment by ursenzler** [comment]
Fixed in 1.0.36
